taichi/backends/opengl/opengl_api.h

```diff
@@ -20,6 +21,9 @@ int opengl_get_threads_per_group();
 #include "taichi/inc/opengl_extension.inc.h"
 #undef PER_OPENGL_EXTENSION
 
+using _RangeSizeEvaluator = std::function<size_t(const char *)>;
+using RangeSizeEvaluator = std::optional<_RangeSizeEvaluator>;
```

Comment:
Actually, `std::function` already includes the optional semantics. You can set it to `nullptr` to indicate there is no `rse`.